test/2_test_fullpath_with_permit.ts

```diff
@@ -5252,6 +5252,205 @@ describe('Cashier and VoucherKernel', () => {
+          await advanceTimeSeconds(60);
+          await utils.finalize(tokenVoucherIdCommited, users.deployer.signer);
+          await utils.finalize(tokenVoucherIdRedeemed, users.deployer.signer);
+          await utils.finalize(tokenVoucherIdRefunded, users.deployer.signer);
+
+          const oldOwnerFromEscrow = await contractCashier.getEscrowAmount(
+            users.other1.address
+          );
+
+          const oldOwnerBalance = await ethers.provider.getBalance(
+            users.other1.address
+          );
+
+          await utils.withdraw(tokenVoucherIdCommited, users.deployer.signer);
+          let expectedBalance = oldOwnerBalance.add(
+            constants.PROMISE_DEPOSITSE1
+          );
+          expect(
+            await contractCashier.getEscrowAmount(users.other1.address)
+          ).to.equal(
+            oldOwnerFromEscrow.sub(constants.PROMISE_DEPOSITSE1),
+            'Wrong escrow amount after expiry'
+          );
+          expect(
+            await ethers.provider.getBalance(users.other1.address)
+          ).to.equal(expectedBalance, 'Wrong withdrawn amount after expiry');
+
+          await utils.withdraw(tokenVoucherIdRedeemed, users.deployer.signer);
+          expectedBalance = expectedBalance.add(
+            constants.PROMISE_DEPOSITSE1 + constants.PROMISE_PRICE1
+          );
+          expect(
+            await contractCashier.getEscrowAmount(users.other1.address)
+          ).to.equal(
+            oldOwnerFromEscrow.sub(2 * constants.PROMISE_DEPOSITSE1),
+            'Wrong escrow amount after redeem'
+          );
+          expect(
+            await ethers.provider.getBalance(users.other1.address)
+          ).to.equal(expectedBalance, 'Wrong withdrawn amount after redeem');
+
+          await utils.withdraw(tokenVoucherIdRefunded, users.deployer.signer);
+          expectedBalance = expectedBalance.add(constants.PROMISE_DEPOSITSE1);
+          expect(
+            await contractCashier.getEscrowAmount(users.other1.address)
+          ).to.equal(
+            oldOwnerFromEscrow.sub(3 * constants.PROMISE_DEPOSITSE1),
+            'Wrong escrow amount after refund'
+          );
+          expect(
+            await ethers.provider.getBalance(users.other1.address)
+          ).to.equal(expectedBalance, 'Wrong withdrawn amount after refund');
+
+          // cancel or fault of remaining voucherset
+          expect(
+            await contractBosonRouter
+              .connect(users.other2.signer)
+              .requestCancelOrFaultVoucherSet(tokenSupplyKey)
+          )
+            .to.emit(contractVoucherKernel, eventNames.LOG_CANCEL_VOUCHER_SET)
+            .withArgs(tokenSupplyKey, users.other2.address);
+
+          expect(
+            await contractCashier.getEscrowAmount(users.other2.address)
+          ).to.equal('0');
+        });
+
+        it('After transfer, old seller cannot manage new vouchers anymore', async () => {
```

Comment:
I would remove the "anymore" here since the old seller should never have been able to manage new vouchers from an already transferred voucher set.

Reply:
:+1: will change in next commit.